lib/looker-sdk/client.rb

```diff
@@ -238,6 +238,32 @@ class Serializer < Sawyer::Serializer
       def encode(data)
         data.kind_of?(Faraday::UploadIO) ? data : super
       end
+
+      # slight modification to the base class' decode_has_value function to
+      # less permissive when decoding time values.
+      #
+      # See https://github.com/looker/helltool/issues/22037 for more details
+      def decode_hash_value(key, value)
```

Comment:
I see: https://github.com/lostisland/sawyer/blob/6eced4624217eb65d8c90be9d81b11c0bebd81e4/lib/sawyer/serializer.rb#L101

Given that your method is mostly like the original, I'd be inclined to change this to a snippet that used `alias` to remember the original, and then just did something more like...

```ruby
if time_field?(key, value) && value.is_a?(String)
  begin
    Time.iso8601(value)
  rescue ArgumentError
    value
  end
else
  original_decode_hash_value(key, value)
end
```